### PR TITLE
[NO QA]Add a try/catch for build in review error for iOS

### PR DESCRIPTION
**Pull request by AndrewGable**
### Details
Adds a try/catch for the error message we were receiving for the iOS errors.

### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/178492

### Tests
1. Merge this PR
2. Verify we do not see iOS build errors like https://github.com/Expensify/App/runs/3747787438

### QA Steps
N/A

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -111,23 +111,31 @@ platform :ios do
...
+              demo_account_name: ENV["APPLE_DEMO_EMAIL"],
+              demo_account_password: ENV["APPLE_DEMO_PASSWORD"],
+              notes: "Use the account provided. Thank you for the review."
+          }
+      )
+    rescue Exception => e
+      if e.message.include? "Another build is in review"
+        UI.important("Another build is already in external beta review. Skipping external beta review submission")
+      else
+        raise
```
Comment (Jag96) [inline fastlane/Fastfile:136]: nab - Can we do something like this instead to remove the if/else?
```
  rescue Exception => e
    raise e unless e.message.include? "Another build is in review"
    UI.important("Another build is already in external beta review. Skipping external beta review submission")
  end
```

**Review by Jag96 — COMMENTED** [review]
LGTM, just 1 nab question

File: fastlane/Fastfile (hunk 1 above)
Comment (AndrewGable) [reply fastlane/Fastfile:136]: Yes, I think so- Is that valid ruby?

Comment (Jag96) [reply fastlane/Fastfile:136]: I think that'll work, but I'm fine leaving it simple w/ the if/else since its more straightforward

**Review by Jag96 — APPROVED** [review]

**Pull request merged**